TypeCobol/Compiler/Symbols/TypedefSymbol.cs

```diff
@@ -0,0 +1,272 @@
+﻿using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Scopes;
+
+namespace TypeCobol.Compiler.Symbols
+{
+    /// <summary>
+    /// A Typedef Symbol
+    /// </summary>
+    public class TypedefSymbol : VariableSymbol, IDomain<VariableSymbol>
+    {
+        /// <summary>
+        /// Named constructor.
+        /// </summary>
+        /// <param name="name"></param>
+        public TypedefSymbol(String name)
+            : base(name)
+        {
+            base.Kind = Kinds.Typedef;
+        }
+        /// <summary>
+        /// The Domain of this program.
+        /// </summary>
+        internal Dictionary<string, Scope<VariableSymbol>.MultiSymbols> Domain
```

Comment:
Ok this is in order to be compatible with the current TypeCobol which lookup symbol references into TypeDef declarations, but when we will switch to this implementation, it will not be necessary, but it can be used to lookup symbol inside a TypeDef if necessary